File: app/src/main/java/com/sakuna63/tumbin/application/contract/PostsContract.java
package com.sakuna63.tumbin.application.contract;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import com.sakuna63.tumbin.data.model.Post;

import java.util.List;

import io.realm.RealmResults;

public interface PostsContract {
    interface View extends BaseView<Presenter> {
        void showPosts(@NonNull List<Post> posts);

        void hidePosts();

        void showError(@Nullable String message);

        void hideError();

        void showEmpty();

        void hideEmpty();

        void showLoading();

        void hideLoading();

        void showPostDetail(@NonNull Post post);

        void scrollTo(long postId, int offsetPx);

        int getScrollY();
    }

    interface Presenter extends BasePresenter {
        void onPostClick(@NonNull Post post);

        void onReloadClick();

        void onRefresh();

        void onScrollBottom();
    }
}

File: app/src/main/java/com/sakuna63/tumbin/application/contract/PostsContract.kt
package com.sakuna63.tumbin.application.contract

import com.sakuna63.tumbin.data.model.Post

interface PostsContract {
    interface View : BaseView<Presenter> {
        fun showPosts(posts: List<Post>)

        fun hidePosts()

        fun showError(message: String)

        fun hideError()

        fun showEmpty()

        fun hideEmpty()

        fun showLoading()

        fun hideLoading()

        fun showPostDetail(post: Post)

        fun scrollTo(postId: Long, offsetPx: Int)

        val scrollY: Int
    }

    interface Presenter : BasePresenter {
        fun onPostClick(post: Post)

        fun onReloadClick()

        fun onRefresh()

        fun onScrollBottom()
    }
}
